lib/puppet/indirector/terminus.rb

```diff
@@ -66,18 +66,18 @@ def inherited(subclass)
       # Yay, class/instance confusion.
       subclass.terminus_type = self.name
 
-      # Our subclass is specifically associated with an indirection.
+      # This subclass is specifically associated with an indirection.
       raise("Invalid name #{longname}") unless names.length > 0
-      indirection_name = names.pop.sub(/^[A-Z]/) { |i| i.downcase }.gsub(/[A-Z]/) { |i| "_#{i.downcase}" }.intern
+      processed_name = names.pop.sub(/^[A-Z]/) { |i| i.downcase }.gsub(/[A-Z]/) { |i| "_#{i.downcase}" }
 
-      if indirection_name == "" or indirection_name.nil?
+      if processed_name == "" || processed_name.nil?
```

Comment:
Neither `String#sub` nor `String#gsub` will never return `nil` so test for nil can be removed. Suggest `processed_name.empty?`